### PR TITLE
API-1565: add "is_root" search filter

### PR DESCRIPTION
**Pull request by pchasle**
Add documentation of the filter to get root or non-root categories only.

### CODE REVIEW

**Review by tomglvng — APPROVED** [review]

File: content/rest-api/filter.md [hunk 1]
```diff
@@ -1217,3 +1217,19 @@ Below you will find the operator allowed to filter on this property, as well as
 ```
 /api/rest/v1/categories?search={"code":[{"operator":"IN","value":["category_code1","category_code2"]}]}
 ```
+
+### By root category
+::: availability versions=Serenity editions=CE,EE
+
+You can filter the categories to get only root categories or non-root categories.
```
Comment (meganelepalud-akeneo) [inline content/rest-api/filter.md:1224]:
```suggestion
You can filter the categories to get only root categories or non-root categories. Root categories can also be called Category trees. 
```

File: content/rest-api/filter.md [hunk 2]
```diff
@@ -1217,3 +1217,19 @@ Below you will find the operator allowed to filter on this property, as well as
...
+
+Below you will find the operator allowed to filter on this property, as well as the corresponding type of value required in the `search` query parameter.
+
+| Operator | Allowed value type | Filter description |
+| ----------------- | -------------- | ------------------ |
+| `=` | boolean | Only returns categories that are root (`true`) or non-root (`false`) |
+
+#### Example
+```
+/api/rest/v1/categories?search={"IS_ROOT":[{"operator":"=","value":true}]}
```
Comment (meganelepalud-akeneo) [inline content/rest-api/filter.md:1234]:
```suggestion
/api/rest/v1/categories?search={"is_root":[{"operator":"=","value":true}]}
```
I tried with IS_ROOT and I have an error message: "Available search filters are \"code, parent, is_root\" and you tried to search on unavailable filter \"IS_ROOT\""

File: content/rest-api/filter.md [hunk 3]
```diff
@@ -1217,3 +1217,19 @@ Below you will find the operator allowed to filter on this property, as well as
...
+### By root category
+::: availability versions=Serenity editions=CE,EE
+
+You can filter the categories to get only root categories or non-root categories.
+
+Below you will find the operator allowed to filter on this property, as well as the corresponding type of value required in the `search` query parameter.
+
+| Operator | Allowed value type | Filter description |
+| ----------------- | -------------- | ------------------ |
+| `=` | boolean | Only returns categories that are root (`true`) or non-root (`false`) |
```
Comment (meganelepalud-akeneo) [inline content/rest-api/filter.md:1230]:
```suggestion
| `=` | boolean | When `true`, only returns categories that are root. When `false`, returns any other categories. |
```

File: content/rest-api/filter.md [hunk 4]
```diff
@@ -1217,3 +1217,19 @@ Below you will find the operator allowed to filter on this property, as well as
...
 /api/rest/v1/categories?search={"code":[{"operator":"IN","value":["category_code1","category_code2"]}]}
 ```
+
+### By root category
+::: availability versions=Serenity editions=CE,EE
+
+You can filter the categories to get only root categories or non-root categories.
+
+Below you will find the operator allowed to filter on this property, as well as the corresponding type of value required in the `search` query parameter.
+
```
Comment (meganelepalud-akeneo) [inline content/rest-api/filter.md:1227]:
```suggestion

::: tips
You can combine the `is_root` filter with any other filter available on categories.
:::

```

**Review by meganelepalud-akeneo — CHANGES_REQUESTED** [review]

**Comment by meganelepalud-akeneo** [comment]
I have a meeting with Camille tomorrow, I'll show her this PR. ;)

File: content/rest-api/filter.md [hunk 5]
```diff
@@ -1217,3 +1217,23 @@ Below you will find the operator allowed to filter on this property, as well as
 ```
 /api/rest/v1/categories?search={"code":[{"operator":"IN","value":["category_code1","category_code2"]}]}
 ```
+
+### By root category
+::: availability versions=Serenity editions=CE,EE
+
+You can filter the categories to get only root categories or non-root categories. Root categories can also be called Category trees.
```
Comment (fitn) [inline content/rest-api/filter.md:1224]:
```suggestion
You can filter the categories to get only root categories or non-root categories.
```

A root category is the base of the category tree. The category tree is a root category + some categories

**Review by fitn — APPROVED** [review]

File: content/rest-api/filter.md [hunk 6]
```diff
@@ -1217,3 +1217,23 @@ Below you will find the operator allowed to filter on this property, as well as
 ```
 /api/rest/v1/categories?search={"code":[{"operator":"IN","value":["category_code1","category_code2"]}]}
 ```
+
+### By root category
```
Comment (tseho) [inline content/rest-api/filter.md:1221]: `By root category` is a little confusing for me.
At first glance, I think it means that I'm able to filter on a root category.
Something like `?root=clothes` and receives all the categories of the tree under this root.

Maybe `By root only` ?

**Review by camillebourjade — APPROVED** [review]

**Review by meganelepalud-akeneo — APPROVED** [review]

**Pull request merged**